executorCommon/src/main/scala/dx/executor/WorkflowExecutor.scala

```diff
@@ -198,7 +199,23 @@ abstract class WorkflowExecutor[B <: Block[B]](jobMeta: JobMeta, separateOutputs
   }
 
   private def getAnalysisOutputFiles(analysis: DxAnalysis): Map[String, DxFile] = {
-    val desc = analysis.describe(Set(Field.Input, Field.Output))
+    // the analysis is updated asynchronously, so we need to check the dependsOn field
+    // to determine if it is still waiting on the output of a previous stage before we
+    // can proceed with the reorg
+    val desc = Iterator
+      .continually(
+          analysis.describeNoCache(Set(Field.Input, Field.Output, Field.DependsOn))
+      )
+      .collectFirstDefined { a =>
+        a.dependsOn match {
+          case Some(Vector(jobId)) if jobId == jobMeta.jobId => Some(a)
+          case None | Some(Vector())                         => Some(a)
+          case _ =>
+            Thread.sleep(3000)
```

Comment:
Cant we get stuck in infinite loop in case jobId wouldn't change for some reason? Or is it impossible not to pass the jobId == jobMeta.jobId condition in the end?

I was wondering whether we want to have some kind of timeout here - just to be safe

Reply:
Yes it's possible. I asked this question in GChat and Stanley commented that the job will eventually terminate due to timeout, so it's ok to retry without a limit. I still think a max number of retries is a good idea. I'll bring it up at stand-up and see what Eugene thinks.